src/Context/DefaultContextTrait.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Sentry\Context;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+trait DefaultContextTrait
+{
+    /**
+     * @var OptionsResolver The options resolver
+     */
+    private $resolver;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function __construct(array $data = [])
+    {
+        $this->resolver = new OptionsResolver();
+
+        $this->configureOptions($this->resolver);
+
+        parent::__construct($this->resolver->resolve($data));
```

Comment:
I don't like this trait a lot, because it calls `parent` many times, so it's making assumptions.

Maybe here we're missing an `abstract` intermediate class?

Reply:
Or even we can move this to the base `Context` class?

Reply:
Just a note: I didn't put this in the base `Context` class because not all contexts need validation and it would be useless to use the `OptionsResolver` component in those cases